crates/mdb/src/lib.rs

```diff
@@ -207,7 +207,17 @@ impl SubscriberConfig {
         debug!("Creating {}...", any::type_name::<Self>());
         unsafe {
             let raw_on_message = Box::into_raw(Box::new(on_message));
-            let on_message = Deferred::new(raw_on_message);
+            // TODO: Verify that panic is sound.
+            // SAFETY: There a few ways this can be dropped:
+            // * This function panics; since the user doesn't get a config back the config either
+            //   is leaked or it wasn't created. In either case the pointer will never be
+            //   dereferenced.
+            // * The struct returned from this function is not used; since the callback is dropped
+            //   after the drop implementation for this type this is sound even if drop would
+            //   dereference the pointer, which I don't think it does.
```

Comment:
which I don't think it does -> which it doesn't. The comment in `Drop` says it doesn't dereference the pointer, so should this.